mongo/options/clientoptions_test.go

```diff
@@ -1247,6 +1264,48 @@ func TestSetURIopts(t *testing.T) {
 			},
 			wantErrs: nil,
 		},
+		{
+			name: "ALLOWED_HOSTS cannot be specified in URI connection",
+			uri:  "mongodb://localhost/?authMechanism=MONGODB-OIDC&authMechanismProperties=ALLOWED_HOSTS:example.com",
+			wantopts: &ClientOptions{
+				HTTPClient: httputil.DefaultHTTPClient,
+			},
+			wantErrs: []error{
+				errors.New(`error validating uri: ALLOWED_HOSTS cannot be specified in the URI connection string for the "MONGODB-OIDC" auth mechanism, it must be specified through the ClientOptions directly`),
+			},
+		},
+		{
+			name: "colon in TOKEN_RESOURCE works as expected",
+			uri:  "mongodb://example.com/?authMechanism=MONGODB-OIDC&authMechanismProperties=TOKEN_RESOURCE:mongodb://test-cluster",
+			wantopts: &ClientOptions{
+				Hosts:      []string{"example.com"},
+				Auth:       &Credential{AuthMechanism: "MONGODB-OIDC", AuthSource: "$external", AuthMechanismProperties: map[string]string{"TOKEN_RESOURCE": "mongodb://test-cluster"}},
+				HTTPClient: httputil.DefaultHTTPClient,
+			},
+			wantErrs: nil,
+		},
+		{
+			name: "tmp",
+			uri:  "mongodb://example.com/?authMechanism=MONGODB-OIDC&authMechanismProperties=TOKEN_RESOURCE:mongodb://test-cluster,ENVIRONMENT:azureManagedIdentities",
+			wantopts: &ClientOptions{
+				Hosts: []string{"example.com"},
+				Auth: &Credential{AuthMechanism: "MONGODB-OIDC", AuthSource: "$external", AuthMechanismProperties: map[string]string{
+					"ENVIRONMENT":    "azureManagedIdentities",
+					"TOKEN_RESOURCE": "mongodb://test-cluster"}},
+				HTTPClient: httputil.DefaultHTTPClient,
+			},
+			wantErrs: nil,
+		},
+		{
+			name: "comma in key:value pair causes error",
+			uri:  "mongodb://example.com/?authMechanismProperties=TOKEN_RESOURCE:mongodb://host1%2Chost2",
+			wantopts: &ClientOptions{
+				HTTPClient: httputil.DefaultHTTPClient,
+			},
+			wantErrs: []error{
+				errors.New(`rror parsing uri: invalid authMechanism property`),
```

Comment:
Seems like the test failures are due to a typo here.
```suggestion
				errors.New(`error parsing uri: invalid authMechanism property`),
```